File: pythonApi/uiService.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import requests

# URL of the game service endpoint
GAME_SERVICE_URL = 'http://localhost:5001/play'

app = Flask(__name__)
CORS(app)  # Enable CORS for all routes in the app

@app.route('/play', methods=['POST'])
def play():
    # Example: Receive player's move from the request
    player_move = request.json.get('move')
    try:
        # JSON payload to send to the game service
        payload = {'move': player_move}

        # Make a POST request to the game service
        response = requests.post(GAME_SERVICE_URL, json=payload)

        # Check if the request was successful (status code 200)
        if response.status_code == 200:
            # Parse the JSON response from the game service
            result = response.json().get('result')
            return jsonify({'result': result})
        else:
            # Log or handle the error appropriately
            print(f'Error: HTTP {response.status_code} - {response.text}')
            return None
    except Exception as e:
        # Log or handle the exception appropriately
        print(f'Error: {e}')
        return None

if __name__ == '__main__':
    app.run(debug=True)

File: nodeApi/uiService.js
const express = require('express');
const cors = require('cors');
const axios = require('axios');
const { GAME_SERVICE_URL } = require('./constants');

const app = express();
app.use(express.json());
app.use(cors()); // Enable CORS for all routes in the app

app.post('/play', async (req, res) => {
    try {
        // Receive player's move from the request
        const { player } = req.body;

        // JSON payload to send to the game service
        const payload = { player };

        // Make a POST request to the game service
        const response = await axios.post(GAME_SERVICE_URL, payload);

        // Check if the request was successful
        if (response.status === 200) {
            const result = response.data;
            res.json(result);
        } else {
            // Log error
            console.error(`Error: HTTP ${response.status} - ${response.statusText}`);
            res.sendStatus(response.status);
        }
    } catch (error) {
        console.error(`Error: ${error.message}`);
        res.sendStatus(500);
    }
});

const startUiService = () => {
    const PORT = process.env.PORT || 5000;
    app.listen(PORT, () => {
        console.log(`Server is running on port ${PORT}`);
    });
}

module.exports = { startUiService };